app/uk/gov/hmrc/exports/models/declaration/submissions/Action.scala

```diff
@@ -18,17 +18,69 @@ package uk.gov.hmrc.exports.models.declaration.submissions
 
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
+import uk.gov.hmrc.exports.models.declaration.ExportsDeclaration
+import uk.gov.hmrc.exports.models.declaration.submissions.RequestType.RequestTypeFormat
 
 import java.time.{LocalDateTime, ZoneId, ZonedDateTime}
 
-case class Action(
-  id: String,
-  requestType: RequestType,
-  requestTimestamp: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")),
-  notifications: Option[Seq[NotificationSummary]] = None
-) {
+sealed trait Action {
+
+  val id: String
+  val notifications: Option[Seq[NotificationSummary]]
+  val requestTimestamp: ZonedDateTime
+
   val latestNotificationSummary: Option[NotificationSummary] =
     notifications.flatMap(_.lastOption)
+
+  val versionNo: Int
+
+}
+
+case class SubmissionAction(
+  id: String,
+  requestTimestamp: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")),
+  notifications: Option[Seq[NotificationSummary]] = None,
+  decId: String
+) extends Action {
+  val versionNo: Int = 1
+}
+
+case class CancellationAction(
+  id: String,
+  requestTimestamp: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")),
+  notifications: Option[Seq[NotificationSummary]] = None,
+  decId: String,
+  versionNo: Int
+) extends Action
+
+object CancellationAction {
+  def apply(id: String, submission: Submission) =
+    new CancellationAction(id, decId = submission.latestDecId, versionNo = submission.latestVersionNo)
+}
+
+case class AmendmentAction(
+  id: String,
+  requestTimestamp: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")),
+  notifications: Option[Seq[NotificationSummary]] = None,
+  decId: String,
+  versionNo: Int
+) extends Action
+
+object AmendmentAction {
+  def apply(id: String, declaration: ExportsDeclaration, submission: Submission) =
+    new AmendmentAction(id, decId = declaration.id, versionNo = submission.latestVersionNo + 1)
+}
+
+case class ExternalAmendmentAction(
```

Comment:
Wasn't there the idea that `decId` would get populated for this class after it's been received and parsed by some job? (can't quite remember now)

Reply:
Good question, this needs clarity

Reply:
Indeed you're right, will make the change